src/jobservice/job/impl/notification/webhook_job.go

```diff
@@ -0,0 +1,100 @@
+package notification
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/goharbor/harbor/src/common/utils/registry"
+	"github.com/goharbor/harbor/src/jobservice/job"
+	"github.com/goharbor/harbor/src/jobservice/logger"
+	"net/http"
+	"os"
+	"strconv"
+)
+
+// Max retry has the same meaning as max fails.
+const maxFails = "JOBSERVICE_WEBHOOK_JOB_MAX_RETRY"
+
+// WebhookJob implements the job interface, which send notification by http or https.
+type WebhookJob struct {
+	client *http.Client
+	logger logger.Interface
+	ctx    job.Context
+}
+
+// MaxFails returns that how many times this job can fail, get this value from ctx.
+func (wj *WebhookJob) MaxFails() uint {
+	if maxFails, exist := os.LookupEnv(maxFails); exist {
+		result, err := strconv.ParseUint(maxFails, 10, 32)
+		// Unable to log error message because the logger isn't initialized when calling this function.
+		if err == nil {
+			return uint(result)
+		}
+	}
+
+	// Default max fails count is 10, and its max retry interval is around 3h
+	// Large enough to ensure most situations can notify successfully
+	return 10
+}
+
+// ShouldRetry ...
+func (wj *WebhookJob) ShouldRetry() bool {
+	return true
+}
+
+// Validate implements the interface in job/Interface
+func (wj *WebhookJob) Validate(params job.Parameters) error {
+	return nil
+}
+
+// Run implements the interface in job/Interface
+func (wj *WebhookJob) Run(ctx job.Context, params job.Parameters) error {
+	if err := wj.init(ctx, params); err != nil {
+		return err
+	}
+
+	err := wj.execute(ctx, params)
+	return err
+}
+
+// init webhook job
+func (wj *WebhookJob) init(ctx job.Context, params map[string]interface{}) error {
+	wj.logger = ctx.GetLogger()
+	wj.ctx = ctx
+
+	// default insecureSkipVerify is true
+	insecureSkipVerify := true
+	if v, ok := params["skip_cert_verify"]; ok {
+		insecureSkipVerify = v.(bool)
```

Comment:
if !ok {
  return err？
}

Reply:
skip_cert_verify isn't a mandatory parameter, the default value should be taken instead of returning an error when it does not exist in params.

Reply:
There is a must-fix we ignored, that the default value of `insecureSkipVerify ` has to be `false`, I was educated by security guys more than a couple of times...

Reply:
@lightof410
You updated the value but the comment still says "// default insecureSkipVerify is true" !!!